### DISCUSSION TITLE
<template x-if="..."> only renders the first child

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3501] [4. General]
I have two modes in my navigation. Since I didn't want to have `x-show` in every element, the idea was to use a countainer tag. In order to not have too many `<div>` tags around that say nothing (semantically), I was trying to use `<template x-if="...">` for all the `<button>` elements inside the `<nav>`. However, I see now, that only the first button gets rendered. I assume this works as designed?

Wouldn't it be better if all the elements inside would get rendered?

**Comment by danddanddand** [comment]
> template tags can only contain one root level element. 

[https://alpinejs.dev/directives/if](https://alpinejs.dev/directives/if)

**Comment by simonferndriger** [comment]
Also, using an `x-ref` inside the `<template>` doesn't seem to work?

**Comment by danddanddand** [comment]
If the x-if condition isn't true, it's essentially not in DOM tree so x-ref or querySelector or whatever wouldn't work. If it's true, and in DOM tree therefore, I expect x-ref would work but you'd need to show your code for me to understand.

**Comment by ekwoka** [comment]
As mentioned, Template tags can only have one child to work with how Alpine handles tracking the generated elements.

Second, as mentioned, x-refs can only work while the element is present in the dom. If the condition creating the element is not true, the element won't exist.

If it was true, and is now false, the x-ref may be defined, but incorrectly to the disconnected element.

**Comment by simonferndriger** [comment]
OK, thank, that makes sense. It solved the issue for me, so I'm closing this. However, absolutely great framework, I love it!

**Comment by Stetzon** [comment]
This one just got me for a good hour.  Would it be possible to add a warning to the console when accidentally placing multiple root elements under the template tag?  Vue(2) would do this for their SFCs

**Reply by ekwoka** [reply]
🤔

**Comment by jay8anks** [comment]
Got hit by this one as well. I will say that an Alpinejs "repeater," that just repeated whatever markup was in a template would be a handy feature. 

-=-=

Edit: Well, on messing around with this some more, I was pretty much able to repeat as many elements as I wanted by putting them all in a parent div. Might not be ideal, but preliminary results seem to work.
